nixos/modules/services/networking/iwd.nix

```
@@ -48,33 +48,49 @@ in
     };
   };
 
-  config = mkIf cfg.enable {
-    assertions = [{
-      assertion = !config.networking.wireless.enable;
-      message = ''
-        Only one wireless daemon is allowed at the time: networking.wireless.enable and networking.wireless.iwd.enable are mutually exclusive.
-      '';
-    }];
-
-    environment.etc."iwd/${configFile.name}".source = configFile;
-
-    # for iwctl
-    environment.systemPackages = [ cfg.package ];
-
-    services.dbus.packages = [ cfg.package ];
-
-    systemd.packages = [ cfg.package ];
-
-    systemd.network.links."80-iwd" = {
-      matchConfig.Type = "wlan";
-      linkConfig.NamePolicy = "keep kernel";
-    };
+  config = lib.mkMerge [
+    (mkIf cfg.enable {
+      assertions = [{
+        assertion = !config.networking.wireless.enable;
+        message = ''
+          Only one wireless daemon is allowed at the time: networking.wireless.enable and networking.wireless.iwd.enable are mutually exclusive.
+        '';
+      }];
+
+      environment.etc."iwd/${configFile.name}".source = configFile;
+
+      # for iwctl
+      environment.systemPackages = [ cfg.package ];
+      services.dbus.packages = [ cfg.package ];
+      systemd.packages = [ cfg.package ];
+
+      systemd.network.links."80-iwd" = {
+        matchConfig.Type = "wlan";
+        linkConfig.NamePolicy = "keep kernel";
+      };
 
-    systemd.services.iwd = {
-      wantedBy = [ "multi-user.target" ];
-      restartTriggers = [ configFile ];
-    };
-  };
+      systemd.services.iwd = {
+        wantedBy = [ "multi-user.target" ];
+        restartTriggers = [ configFile ];
+      };
+    })
+
+    (
+      let
+        needResolv = cfg.enable
+          && lib.hasAttrByPath [ "Network" "NameResolvingService" ] cfg.settings
+          && cfg.settings.Network.NameResolvingService == "resolvconf";
+      in
+      mkIf needResolv {
+        environment.systemPackages = [ pkgs.openresolv ];
+
+        systemd.services.iwd = {
+          path = [ pkgs.openresolv ];
+          serviceConfig.ReadWritePaths = "/etc/resolv.conf";
```

```suggestion
          serviceConfig.ReadWritePaths = "-/etc/resolv.conf";
```

In case it does not exist.